modules/rte_mis_core/src/Plugin/Block/RoleBasedDetailsBlock.php

```diff
@@ -0,0 +1,599 @@
+        ->getQuery()
+        ->condition('roles', 'school')
+        ->condition('status', 1)
+        ->accessCheck(FALSE);
+      $school_ids = $query->execute();
+
+      // Return 0 if no schools found.
+      if (empty($school_ids)) {
+        return 0;
+      }
+
+      // Step 2: Filter schools by approval status.
+      $query = $this->entityTypeManager->getStorage('user')
+        ->getQuery()
+        ->condition('uid', $school_ids, 'IN')
+        ->accessCheck(FALSE)
+        ->condition('field_school_details.entity:mini_node.field_school_verification', $verification_status);
+      $approved_nids = $query->execute();
+
+      // Count the approved schools.
+      if (!empty($approved_nids)) {
+        $approved_count = count($approved_nids);
+      }
+
+      return $approved_count;
+    }
+
+    if ($current_role === 'district_admin' || $current_role === 'block_admin') {
+      $matchingSchoolIds = $this->gettingMatchingSchools();
+
+      // Step 2: Filter schools by approval status.
+      if (!empty($matchingSchoolIds)) {
+        $query = $this->entityTypeManager->getStorage('user')
+          ->getQuery()
+          ->condition('uid', $matchingSchoolIds, 'IN')
+          ->accessCheck(FALSE)
+          ->condition('field_school_details.entity:mini_node.field_school_verification', $verification_status);
+        $approved_nids = $query->execute();
+
+        // Count the approved schools.
+        if (!empty($approved_nids)) {
+          $approved_count = count($approved_nids);
+        }
+      }
+
+      return $approved_count;
+    }
+
+    // Return 0 if $current_role doesn't match any condition.
+    return 0;
+  }
+
+  /**
+   * Gets the count of rejected schools based on the user's role.
+   *
+   * @param string $current_role
+   *   The role of the user ('state_admin', 'district_admin', 'block_admin').
+   *
+   * @return int
+   *   The count of rejected schools based on the user's role.
+   */
+  public function getSchoolRejectCount($current_role): int {
+    $verification_status = 'school_registration_verification_rejected';
+    $rejected_count = 0;
+    $matchingSchoolIds = [];
+
+    if ($current_role === 'state_admin') {
+      // Step 1: Query all users with role 'school'.
+      $query = $this->entityTypeManager->getStorage('user')
+        ->getQuery()
+        ->condition('roles', 'school')
+        ->condition('status', 1)
+        ->accessCheck(FALSE);
+      $school_ids = $query->execute();
+
+      // Return 0 if no schools found.
+      if (empty($school_ids)) {
+        return 0;
+      }
+
+      // Step 2: Filter schools by rejection status.
+      $query = $this->entityTypeManager->getStorage('user')
+        ->getQuery()
+        ->condition('uid', $school_ids, 'IN')
+        ->accessCheck(FALSE)
+        ->condition('field_school_details.entity:mini_node.field_school_verification', $verification_status);
+      $rejected_nids = $query->execute();
```

Comment:
- Can we combine these two query?
- Also consider the school_admin roles. this will be needed in renewal process.